src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipeline.cs

```diff
@@ -35,6 +35,9 @@ internal HttpPipeline(HttpPipelinePolicy[] policies, IServiceProvider services =
             _pipeline = policies;
         }
 
+        public static string ApplicationId { get; set; }
+        public static bool DisableTelemetry { get; set; } = false;
```

Comment:
Not in the scope of these changes, but the name `DisableTelemetry` feels awkward to me, semantically.  Should we consider something like `TelemetryEnabled` or `TelemetryState { Enabled, Disabled }` to avoid the "if not disabled" cognitive burden?

Reply:
Yeah, I agree. We should fix the name or make the API into a method. I would prefer to keep it bool, so that we don't have to add a type.